fsspec/spec.py

```diff
@@ -1134,7 +1135,7 @@ def expand_path(self, path, recursive=False, maxdepth=None, **kwargs):
         if maxdepth is not None and maxdepth < 1:
             raise ValueError("maxdepth must be at least 1")
 
-        if isinstance(path, str):
+        if isinstance(path, (str, os.PathLike[str], pathlib.Path)):
```

Comment:
```suggestion
        if isinstance(path, (str, os.PathLike, pathlib.Path)):
```

Reply:
The [str] form is not allowed on older pythons, causing CI fails. Also, is there any justification of PathLike alongside Path?

Reply:
I just took this over from `stringify_path`, which contains legacy logic to deal with Pythons were Path is not an instance of `PathLike`.